test/distributed/_composable/fsdp/test_fully_shard_training.py

```diff
@@ -52,14 +51,15 @@
 from torch.testing._internal.distributed.checkpoint_utils import with_temp_dir
 
 
-class TestFullyShardForwardInputs(FSDPTestMultiThread):
```

Comment:
Are these changes from MTPG to normal NCCL PG meant for landing? I worry that this will (heavily) increase CI time. We want MTPG for these non-training-numerics related tests.

Reply:
for MTPG, I observed flakiness where dynamo compains about broken process pool. There is a recent post about the issue: https://github.com/pytorch/pytorch/issues/121197. we can discuss if this worth spending time to dig into

Reply:
I wonder if it is worth going through some of these tests and seeing if testing `torch.compile` would be valuable or not. Some of these seem like `torch.compile` for compute should not affect. As a bonus, I would hope we could keep the eager tests to use MTPG somehow 🤔

Reply:
MTPG works well after clearing dynamo cache with ``torch._dynamo.reset()`` before and after each test. this also avoid triggering dynamic shapes